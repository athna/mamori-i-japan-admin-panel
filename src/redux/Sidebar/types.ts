export type SidebarStates = {
  isCollapse: boolean;
};
