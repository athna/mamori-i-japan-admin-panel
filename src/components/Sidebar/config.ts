import { HOST } from '../../constants';


export default [
  {
    name: 'Dashboard',
    icon: 'user',
    path: HOST,
    children: []
  },
  {
    name: 'Restaurant',
    icon: 'user',
    path: HOST + 'restaurant',
    children: []
  },
]