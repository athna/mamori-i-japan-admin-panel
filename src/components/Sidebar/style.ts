import styled from 'styled-components';

export const Logo = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 60px;
  color: #fff;
  font-size: 32px;
  line-height: 1px;

    img {
      flex-shrink: 0;
      width: 32px;
      height: 32px;
      border-radius: 4px;
    }

    p {
      margin: 0;
      padding-left: 8px;
      font-size: 14px;
      font-weight: bold;
    }
`;
