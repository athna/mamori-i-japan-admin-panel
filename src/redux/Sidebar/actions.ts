import { createAction } from 'redux-actions';
import actionTypes from './actionTypes';

export const toggleSidebarCollapseAction = createAction(
  actionTypes.TOGGLE_SIDEBAR_COLLAPSE
);
