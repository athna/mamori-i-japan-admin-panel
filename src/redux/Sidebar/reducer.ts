import { handleActions } from 'redux-actions';
import actionTypes from './actionTypes';

const initialState = {
  isCollapse: false
};

export default handleActions(
  {
    [actionTypes.TOGGLE_SIDEBAR_COLLAPSE]: (state) => ({
      isCollapse: !state.isCollapse
    })
  },
  initialState
);
