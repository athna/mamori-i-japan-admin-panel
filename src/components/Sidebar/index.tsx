import React, { useState, useContext } from 'react';
import { Link } from 'react-router-dom';
import { Menu, Icon, Layout } from 'antd';
import { Logo } from './style';
import config from './config';

import { I18nContext } from '../../locales';

const { Sider } = Layout;

interface SidebarProps {
  sidebarIsCollapse: boolean;
  locale?: any;
}

export default ({ sidebarIsCollapse }: any) => {
  const { langCode } = useContext(I18nContext);
  const [selectedKey, setSelectedKey] = useState('0');

  return (
    <Sider collapsed={sidebarIsCollapse}>
      <Logo>Logo</Logo>
      {/* TODO: change '0' to a sdynamic key in sotre */}
      <Menu selectedKeys={[selectedKey]} theme="dark" mode="inline">
        {config.map((item: any, index) => (
          <Menu.Item key={index}>
            <Link to={item.path}>
              <Icon type={item.icon} />
              <span>{item[`${langCode}Name`]}</span>
            </Link>
          </Menu.Item>
        ))}
      </Menu>
    </Sider>
  );
};
