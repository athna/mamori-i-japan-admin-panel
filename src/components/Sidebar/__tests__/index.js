import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Sidebar from '../';

configure({ adapter: new Adapter() });

describe('Sidebar', () => {
  it('should render correctly', () => {
    const history = createMemoryHistory();

    const component = shallow(
      <Router history={history}>
        <Sidebar />
      </Router>
    );

    expect(component).toMatchSnapshot();
  });
});
