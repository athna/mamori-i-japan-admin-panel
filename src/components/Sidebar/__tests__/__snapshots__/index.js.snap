// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render correctly 1`] = `
<Sider>
  <styled.div>
    Logo
  </styled.div>
  <Menu
    mode="inline"
    selectedKeys={
      Array [
        "0",
      ]
    }
    theme="dark"
  >
    <MenuItem
      key="0"
    >
      <Link
        to="/"
      >
        <Icon
          type="user"
        />
        <span>
          ダッシュボード
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      key="1"
    >
      <Link
        to="/items"
      >
        <Icon
          type="user"
        />
        <span>
          一覧temp
        </span>
      </Link>
    </MenuItem>
  </Menu>
</Sider>
`;
